x/registry/keeper/query_role.go

```diff
@@ -0,0 +1,41 @@
+package keeper
+
+import (
+	"context"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/mycel-domain/mycel/x/registry/types"
+)
+
+func (k Keeper) Role(goCtx context.Context, req *types.QueryRoleRequest) (*types.QueryRoleResponse, error) {
+	var role types.DomainRole
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	dms := strings.Split(req.DomainName, ".")
+	switch len(dms) {
+	case 1: // TLD
+		tld, found := k.GetTopLevelDomain(ctx, dms[0])
+		if !found {
+			return nil, status.Error(codes.NotFound, "domain not found")
```

Comment:
Prefer to use error types and `errors.wrapf`

error type: https://github.com/mycel-domain/mycel/blob/107-query-get-domain-role/x/registry/types/errors.go

sample
https://github.com/mycel-domain/mycel/blob/2a69cc1f5579751fb3500d17a871815dbb6b40f0/x/registry/types/message_withdraw_registration_fee.go#L44